DataStructures.Tests/BinaryTreeTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using DataStructures.Trees;
+using System.Text;
+using Xunit;
+
+namespace DataStructures.Tests
+{
+    public class TreeTests
+    {
+        [Fact]
+        public void Add_value_to_empty_tree()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+
+            // Act
+            tree.Add(4);
+
+            //Assert
+            Assert.Equal(4, tree.Root.Value);
+        }
+
+        [Fact]
+        public void Add_one_Left_and_one_Right_node()
```

Comment:
This is technically incorrect because it is a Binary _Search_ Tree, meaning that only values larger than 8 will go to the right.  If you add 4 and 10 then 8 will have 4 to its left and 10 to its right.